### PR TITLE
fix: session storage retrieval for url messes with the url history

### PR DESCRIPTION
**Pull request by AdityaHegde**
- [x] Since we were redirecting url when visiting the home page. Any saved state would lead to a redirect loop when back button was pressed. Using history.replaceState instead to update the url to fix this.
- [x] Switching between dashboards using the breadcrumbs would pollute the other dashboard's state. Using exploreName from loader function to keep things consistent.
- [x] Disable TTD for dashboards without timeseries. Earlier the button was clickable but nothing would happen. Now there will not be a button at all.
- [x] Figure out the E2E failure
- [x] Load metrics view schema for bookmarks.

### CODE REVIEW

**Comment by ericpgreen2** [comment]
I added a filter, pressed the browser's back button, the filter was removed from the URL, but the filter was not actually removed from the dashboard.

File: web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte
Comment (ericpgreen2) [inline web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte:22]: If I'm following correctly, `partialExploreState` is actually the `urlPartialExploreState`. If so, we shouldn't be passing both `urlPartialExploreState` and `urlSearchForPartial`. We should not pass two props that are "synonyms" (different representations of the same state) – we should pass just one of them.

File: web-common/src/features/dashboards/url-state/convertURLToExplorePreset.ts
Comment (ericpgreen2) [inline web-common/src/features/dashboards/url-state/convertURLToExplorePreset.ts:1]: General reminder that this should be a conversion to `ExploreState` not to `ExplorePreset`

File: web-common/src/features/explores/selectors.ts
Comment (ericpgreen2) [inline web-common/src/features/explores/selectors.ts:130]: Let's be more clear about where this state comes from: `defaultExploreYAMLState`, `exploreYAMLDefaultsState`, `stateFromExploreYAMLDefaults`? We should at least use "default"/"defaults" b/c it's the term used in the Explore YAML.

File: web-common/src/features/explores/selectors.ts
Comment (ericpgreen2) [inline web-common/src/features/explores/selectors.ts:145]: We should not be doing this assignment/synthesis here. We should do it in one central place with the rest of the state synthesis. I think that place is `DashboardURLStateSync.svelte`.

File: web-common/src/features/explores/selectors.ts
Comment (ericpgreen2) [inline web-common/src/features/explores/selectors.ts:152]: If we pass the `stateFromLocalStorage` down with the rest of the partial state objects, then I'd hope we don't need to pass down this boolean.

**Review by ericpgreen2 — CHANGES_REQUESTED** [review]

**Comment by briangregoryholmes** [comment]
Related issue here: https://github.com/rilldata/rill/issues/6248

File: web-common/src/features/dashboards/url-state/convertURLToExplorePreset.ts
Comment (AdityaHegde) [reply web-common/src/features/dashboards/url-state/convertURLToExplorePreset.ts:1]: Yes. This PR is not an exhaustive refactor. Lets do that in a separate PR.

File: web-common/src/features/dashboards/stores/persistent-dashboard-state.ts
Comment (ericpgreen2) [reply web-common/src/features/dashboards/stores/persistent-dashboard-state.ts:1]: Confirmed. Opened a separate issue here: https://github.com/rilldata/rill/issues/6259

Comment (AdityaHegde) [reply web-common/src/features/dashboards/stores/persistent-dashboard-state.ts:1]: Removed

File: web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte
Comment (AdityaHegde) [reply web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte]: Passing down all states to DashboardURLStateSync and handling the merging logic there.

File: web-admin/src/routes/[organization]/[project]/explore/[dashboard]/+layout.ts
Comment (ericpgreen2) [inline web-admin/src/routes/[organization]/[project]/explore/[dashboard]/+layout.ts:47]: There's a request waterfall where `fetchBookmarks` is blocked by `fetchExploreSpec`

File: web-admin/src/routes/[organization]/[project]/explore/[dashboard]/+layout.ts
Comment (ericpgreen2) [inline web-admin/src/routes/[organization]/[project]/explore/[dashboard]/+layout.ts:51]: Let's name this state `bookmarkExploreState` and pass it down with that name until we merge it with the other state sources

File: web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte
Comment (ericpgreen2) [inline web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte:29]: "initExploreState" doesn't specify which source of state it is.

File: web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte
Comment (AdityaHegde) [reply web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte:28]: Lets take that refactor in a future PR. `ExploreState` should inherit from `ExplorePreset` to avoid mapping of individual fields.

File: web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte
Comment (AdityaHegde) [reply web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte:29]: Hmmm this is supposed to be bookmark state on cloud explore, token's state for public URLs and undefined for rill-dev. Not sure what a good name would be to capture this. Adding them explicitly like `bookmarkExploreState` etc doesnt make sense since it is irrelevant on rill-dev.

Comment (ericpgreen2) [reply web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte:24]: I think optional props would be most clear: 
```
export let bookmarkExploreState: Partial<ExploreState> | undefined = undefined;
export let tokenExploreState: Partial<ExploreState> | undefined = undefined;
```

Comment (AdityaHegde) [reply web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte:24]: I have renamed the variables until the call to DashboardURLStateSync. We should avoid having cloud/dev specific login in common code.

File: web-admin/src/routes/[organization]/[project]/explore/[dashboard]/+layout.ts
Comment (AdityaHegde) [reply web-admin/src/routes/[organization]/[project]/explore/[dashboard]/+layout.ts]: Calling them in parallel now.

**Review by ericpgreen2 — CHANGES_REQUESTED** [review]
UXQA:
1. Navigate from the project page to a dashboard, add a filter, click the back button, the filter is _not_ removed. Click the back button again, and you're navigated directly to the project page. [Jam](https://jam.dev/c/b4c2230b-bc64-4ae8-a560-b807a813837a).
2. When I highlight a time range, clicking "back" does not remove it: [Jam](https://jam.dev/c/b866cee9-94a5-4b6a-81d1-6b29ad458211)
3. When I move a highlighted time range to the end of the time series, it gets stuck. I'm unable to move it, and clicking `esc` does not remove it. [Jam](https://jam.dev/c/f7aa3d42-7699-4a1c-8439-f1fd453dfaf7)
4. Go from a small timerange & grain (e.g. Last 14 Days, Daily) to a large one (e.g. All Time, which flips to Monthly by default). The dashboard reflects the new time grain (Monthly), but the URL does not (Daily).
5. In the GitHub Analytics example, there are some commit values that can't be filtered on: [Jam](https://jam.dev/c/fd6031e6-1b11-4bf5-8fd0-33f50c9bde26). (Maybe this fix belongs in a separate PR.)

I found the above issues from playing with this branch for 5-10 minutes. I will stop there, so, after fixing, please do a round of QA yourself.

File: web-common/src/features/entity-management/file-artifact.ts
Comment (AdityaHegde) [reply web-common/src/features/entity-management/file-artifact.ts:77]: We need to get all explores that are referring to a metrics view `exploreResource?.explore?.state?.validSpec?.metricsView` so just the name is not enough.

File: web-common/src/features/dashboards/stores/dashboard-store-defaults.ts
Comment (AdityaHegde) [reply web-common/src/features/dashboards/stores/dashboard-store-defaults.ts:11]: Update the variable names. `getFullInitExploreState` for the function and `partialInitState` for the argument.

File: web-common/src/features/dashboards/url-state/getExploreStateFromSessionStorage.ts
Comment (ericpgreen2) [inline web-common/src/features/dashboards/url-state/getExploreStateFromSessionStorage.ts:48]: We should be getting `ExploreState` from session storage, not `ExplorePreset`

File: web-common/src/features/dashboards/url-state/getExploreStateFromSessionStorage.ts
Comment (ericpgreen2) [inline web-common/src/features/dashboards/url-state/getExploreStateFromSessionStorage.ts:18]: I'd expect to see a call to session storage in this function. Perhaps `getExplorePresetForWebView` (where I see the call to session storage actually happens) is an unnecessary additional layer.

**Review by ericpgreen2 — APPROVED** [review]
I'll approve and merge, so we can get this progress into the `main` branch and take the follow-ups in smaller PRs. Let's please try to work via more focused, smaller-in-scope PRs. Most of the bullet points in the PR's description could've merited its own dedicated PR.

**Pull request merged**
